src/controllers/users.ts

```diff
@@ -3,19 +3,29 @@ import status from 'http-status';
 
 import { User } from '../entities/User';
 import { notFoundError } from '../errors';
+import { createPaginationParams, getValidPage } from '../helpers/pagination.helper';
 import { hashPassword } from '../helpers/password.helper';
 import { mapLoginResponse } from '../mappers/user.mapper';
 import * as userService from '../services/users';
 
-export function getUsers(
+export async function getUsers(
   req: Request,
   res: Response,
   next: NextFunction,
 ): Promise<Response | void> {
-  return userService
-    .findAll()
-    .then((users: User[]) => res.send(users))
-    .catch(next);
+  try {
+    const page = getValidPage(req.query.page as string);
+    const paginationParams = createPaginationParams(page);
+
+    const result = await userService.findAll(paginationParams, undefined);
```

Comment:
it’s not clear why the second argument is explicitly undefined.

If it’s optional, omit it for cleaner code.